crypto/bls/signature_batch.go

```diff
@@ -54,3 +54,86 @@ func (s *SignatureBatch) Copy() *SignatureBatch {
 		Messages:   messages,
 	}
 }
+
+// RemoveDuplicates removes duplicate signature sets from the signature batch.
+func (s *SignatureBatch) RemoveDuplicates() (int, *SignatureBatch) {
+	if len(s.Signatures) == 0 || len(s.PublicKeys) == 0 || len(s.Messages) == 0 {
+		return 0, s
+	}
+	sigMap := make(map[string]int)
+	duplicateSet := make(map[int]bool)
+	for i := 0; i < len(s.Signatures); i++ {
+		currSig := string(s.Signatures[i])
```

Comment:
Fun Go quirk: doing sigMap[string(s.Signatures[i])] actually does 0 allocations compared to declaring currSig := string(s.Signatures[i])